generate_reports.py

```diff
@@ -0,0 +1,365 @@
+from common import get_column_data_from_alias
+from common import get_month_string_from_number
+from constants import TEXT_STYLE_NORMAL, TEXT_STYLE_BOLD
+from constants import TEXT_COLOR_RED, TEXT_COLOR_BLUE
+from constants import TEXT_COLOR_WHITE
+from load_data_from_file import load_month_data, load_year_data
+from WeatherData import WeatherData
+
+
+def change_text_color(text, style=0, color=37, background_color=40):
+    """This functions returns a string whose color is changed
+
+    Arguments:
+        text: str:
+            Text data whose color you want to change
+        style: int:
+            Changes style, default style "Normal"
+        color: int:
+            Changes text color, default color "White"
+        background_color: int:
+            Changes background color, default "Black"
+
+    Returns:
+        Returns the string with colored text
+    """
+
+    try:
+        style = int(style)
+        color = int(color)
+        background_color = int(background_color)
+    except Exception:
+        return ""
+    color_string = "\033[" + str(style) + ";" + str(color) + ";"\
+                   + str(background_color) + "m" + str(text)
+    return color_string
+
+
+def arg_max(list_):
+    """Returns the index of maximum element of list
+
+    Arguments:
+        list_:
+            list whose maximum index is required
+
+    Returns:
+        index: int:
+            The index whose value is maximum
+    """
+
+    if not list_:
+        return None
+    max_index = 0
+    max_value = list_[max_index]
+    for i in range(0, len(list_)):
+        if list_[i] > max_value:
+            max_value = list_[i]
+            max_index = i
+    return max_index
```

Comment:
what is happening in this method? Can we not do this computation with lists, instead of needing a separate function?